src/Maybe/find.spec.js

```diff
@@ -0,0 +1,96 @@
+const test = require('tape')
+
+const isFunction = require('../core/isFunction')
+const constant = require('../combinators/constant')
+const isSameType = require('../core/isSameType')
+const Pred = require('../Pred')
+const find = require('./find')
+const Maybe = require('../core/Maybe')
+const helpers = require('../test/helpers')
+const List = require('../core/List')
+const isNumber = require('../core/isNumber')
+
+const { fromArray } = List
+
+const bindFunc = helpers.bindFunc
+
+test('find is protected from bad fn', t => {
+  const fn = bindFunc(fn => find(fn, []))
+  const err = /find: Pred or a predicate function is required for first argument/
+
+  t.throws(fn(undefined), err, 'throws if fn is undefined')
+  t.throws(fn(null), err, 'throws if fn is null')
+  t.throws(fn(0), err, 'throws if fn is falsey number')
+  t.throws(fn(1), err, 'throws if fn is truthy number')
+  t.throws(fn(NaN), err, 'throws if fn is NaN')
+  t.throws(fn(''), err, 'throws if fn is falsey string')
+  t.throws(fn('string'), err, 'throws if fn is truthy string')
+  t.throws(fn(false), err, 'throws if fn is false')
+  t.throws(fn(true), err, 'throws if fn is true')
+  t.throws(fn({}), err, 'throws if fn is empty POJO')
+  t.throws(fn({ hi: 'there' }), err, 'throws if fn is non-empty POJO')
+
+  t.end()
+})
+
+test('find is protected from bad foldable', t => {
+  const fn = bindFunc(v => find(() => true, v))
+  const err = /find: Foldable required for second argument/
+
+  t.throws(fn(undefined), err, 'throws if foldable is undefined')
+  t.throws(fn(null), err, 'throws if foldable is null')
+  t.throws(fn(0), err, 'throws if foldable is falsey number')
+  t.throws(fn(1), err, 'throws if foldable is truthy number')
+  t.throws(fn(NaN), err, 'throws if foldable is NaN')
+  t.throws(fn(''), err, 'throws if foldable is falsey string')
+  t.throws(fn('string'), err, 'throws if foldable is truthy string')
+  t.throws(fn(false), err, 'throws if foldable is false')
+  t.throws(fn(true), err, 'throws if foldable is true')
+  t.throws(fn({}), err, 'throws if foldable is empty POJO')
+  t.throws(fn({ hi: 'there' }), err, 'throws if foldable is non-empty POJO')
+
+  t.end()
+})
+
+test('find works with predicates', t => {
+  const largeNumber = Pred(isNumber).concat(Pred(x => x > 100))
+  const nonBoolPred =
+    x => Pred(constant(x))
+  const findWrapPred =
+    (fn, foldable) => find(Pred(fn), foldable)
+
+  t.equal(find(largeNumber, [ 10, '12', 150, 200, 2000 ]).option(0), 150, 'works with predicates')
+
+  t.equals(find(nonBoolPred(undefined), []).option('Nothing'), 'Nothing', 'returns false when wrapped function returns undefined')
```

Comment:
This is an interesting test...Do not know if we need to test this, as we would get the same results if we returned `true`, or any value on an empty Array/List. This is more a spec for an empty list.

And because of the truthy falsey bits, you could write a whole grouping of tests on JS values like

```js
t.equals(find(nonBoolPred(0), [ 3, 4 ]).option('Nothing'), 'Nothing', 'returns Nothing with falsey number')
t.equals(find(nonBoolPred(1), [ 3, 4 ]).option('Nothing'), 3, 'returns first value in a Just with truthy number')
```

But really do not think we need this test, or all of the ones I described above, as the specs below cover the context of this function, and the specs for `predOrFunc` cover the truthy/falsey values...So I think we have enough coverage to rule this test out.

Thoughts?

Reply:
Yeah i think it can be removed